pkg/daemon/journal.go

```diff
@@ -0,0 +1,61 @@
+package daemon
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/testground/testground/pkg/api"
+	"github.com/testground/testground/pkg/logging"
+)
+
+func (d *Daemon) getJournalHandler(engine api.Engine) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		log := logging.S().With("req_id", r.Header.Get("X-Request-ID"))
+
+		log.Debugw("handle request", "command", "get journal")
+		defer log.Debugw("request handled", "command", "get journal")
+
+		w.Header().Set("Content-Type", "text/plain")
+
+		taskId := r.URL.Query().Get("task_id")
+		if taskId == "" {
+			fmt.Fprintf(w, "url param `task_id` is missing")
+			return
+		}
+
+		tsk, err := engine.GetTask(taskId)
+		if err != nil {
+			fmt.Fprintf(w, "cannot fetch tsk")
+			return
+		}
+
+		result := decodeResultK8s(tsk.Result)
```

Comment:
Yes, but it will be difficult to get the right abstraction now, without having used a few runners for a while with all the necessary outputs stored in `result`. So I'd rather we make it specific, until we nail the correct abstraction, rather than work with a bad abstraction.

---

I guess once we introduce more runners in TaaS, or once we want to use these dashboards locally, we will have to solve that problem.